services/core-web/src/components/noticeOfWork/applications/process/ProcessPermit.js

```diff
@@ -155,24 +191,46 @@ export class ProcessPermit extends Component {
       });
   };
 
-  withdrawApplication = (values) => {
+  updateApplicationStatus = (values) => {
+    const statusLabel = this.props.noticeOfWorkApplicationStatusOptionsHash[
+      values.now_application_status_code
+    ];
     this.props
       .updateNoticeOfWorkStatus(this.props.noticeOfWork.now_application_guid, {
         ...values,
-        now_application_status_code: "WDN",
+        status_reason: null,
       })
       .then(() => {
         this.props.fetchImportedNoticeOfWorkApplication(
           this.props.noticeOfWork.now_application_guid
         );
         this.props.closeModal();
         notification.success({
-          message: "This application has been withdrawn.",
+          message: `The status has been updated to ${statusLabel}.`,
           duration: 10,
         });
       });
   };
 
+  handleGenerateDocumentFormSubmit = (documentType, values) => {
+    const documentTypeCode = documentType.now_application_document_type_code;
+    const newValues = values;
+    documentType.document_template.form_spec
+      .filter((field) => field.type === "DATE")
+      .forEach((field) => {
+        newValues[field.id] = formatDate(newValues[field.id]);
+      });
+    const payload = {
+      now_application_guid: this.props.noticeOfWork.now_application_guid,
+      template_data: newValues,
+    };
+    this.props.generateNoticeOfWorkApplicationDocument(
+      documentTypeCode,
+      payload,
+      "Successfully Created Document and Attached it to this Notice of Work"
```

Comment:
I fixed the casing in this text to match our other toast messages, if you merge in from develop you should resolve the changes.

"Successfully created document and attached it to Notice of Work"

Reply:
resolved